lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4096,6 +4096,30 @@ LogicalResult FIRRTLLowering::visitStmt(AttachOp op) {
   if (getSingleNonInstanceOperand(op))
     return success();
 
+  // If all operands of the attach are internal to this module (none of them
+  // are ports), then they can all be replaced with a single wire, and we can
+  // delete the attach op.
+  bool isAttachInternalOnly = true;
+  for (auto v : inoutValues) {
+    if (isa<BlockArgument>(v)) {
+      isAttachInternalOnly = false;
+      break;
+    }
+  }
```

Comment:
Nit: A slightly cleaner construction would be to use `llvm::none_of` or similar. I think the following is equivalent:

```c++
bool isAttachInternalOnly = 
  llvm::none_of(inoutValue, [](auto v){ return isa<BlockArgument>(v); });
```

(Who says C++ doesn't have functional programming? 😉 )

Reply:
I love it, very happy to be exposed to as many functional-programming concepts in C++ as possible :)